src/main/java/Foo.java

```diff
@@ -54,9 +48,5 @@ public String getDisplayName() {
       return "Test Foo Task";
     }
 
-    @Override
-    public boolean configure(StaplerRequest staplerRequest, JSONObject json) throws FormException {
```

Comment:
But, alas, you cannot leave it in or you will get a linkage error. Jenkins core _uses_ `json-lib` as part of its public API. You cannot override it willy-nilly.

Reply:
Hm in my real class, aside from minimal repro situation, I have these two lines. I know you moved the private thing to its own class in this PR, but is that actually required based on the changes you made? Cause I really need to do this:

    @Override
    public boolean configure(StaplerRequest staplerRequest, JSONObject json) throws FormException {
      // to persist global configuration information call save()
      endPoint = json.get("endPoint") == null ? null : String.valueOf(json.get("endPoint"));
      save();

      return true; // indicate that everything is good so far
    }

Reply:
(I'm obviously going to try everything except the class removal right now, to see what explodes, if anything, aside from listed best practice)

Reply:
Ah ah ah yeah got the linkage error like you said. So how do I even get around that?!

Reply:
I see, you are implementing global configuration. The best thing is to stop doing this from inside a `Descriptor`, which is semi-deprecated, and use `GlobalConfiguration`. That allows you to pick up the _default implementation_ of `configure(StaplerRequest, JSONObject)` (and so not refer directly to `JSON*` in your plugin in ways that would clash with Jenkins core).

Reply:
Awesome! I'm only on a PC today, without anything setup. Can't wait to try this Monday. From the looks of what you did, this should solve all my problems. Thanks!